common/helpers.go

```diff
@@ -16,118 +16,101 @@ const (
 	HEALTH_CHECK_TYPE_NAME = "SoftLayer_Network_Application_Delivery_Controller_LoadBalancer_Health_Check_Type"
 )
 
-func GetDatacenterByName(client softlayer.Client, name string) (int, error) {
+type lookup func([]byte) (int, error)
+
+func GetIdByName(client softlayer.Client, nameMask string, nameType string, nameTypeGet string, name string, lookupFunc lookup) (int, error) {
```

Comment:
Cool. Good to go. I think I would make this common method lowercase so that it is only shared within the package, but up to you.